src/pymanopt/numerics/core.py

```diff
@@ -12,79 +16,102 @@ def inner(*arguments):
     return inner
 
 
-# TODO:
-#   - np.all
-#   - np.any
-#   - np.arange
-#   - np.arccos
-#   - np.arccosh
-#   - np.arctan
-#   - np.arctanh
-#   - np.argmin
-#   - np.argsort
-#   - np.array
-#   - np.block
-#   - np.conjugate
-#   - np.cos
-#   - np.diag
-#   - np.diagonal
-#   - np.eye
-#   - np.finfo
-#   - np.hstack
-#   - np.iscomplexobj
-#   - np.isnan
-#   - np.isrealobj
-#   - np.linalg.cholesky
-#   - np.linalg.det
-#   - np.linalg.eigh
-#   - np.linalg.inv
-#   - np.linalg.norm
-#   - np.linalg.qr
-#   - np.linalg.solve
-#   - np.linalg.svd
-#   - np.log
-#   - np.logspace
-#   - np.newaxis
-#   - np.ones
-#   - np.pi
-#   - np.polyfit
-#   - np.polyval
-#   - np.prod
-#   - np.random.normal
-#   - np.random.randn
-#   - np.random.uniform
-#   - np.real
-#   - np.sin
-#   - np.sinc
-#   - np.sort
-#   - np.spacing
-#   - np.sqrt
-#   - np.sum
-#   - np.tile
-#   - np.trace
-#   - np.transpose
-#   - np.vectorize
-#   - np.vstack
-#   - np.where
-#   - np.zeros
-#   - scipy.special.comb
-#   - scipy.linalg.solve_continuous_lyapunov
-#   - scipy.linalg.expm
-#   - scipy.linalg.logm
+@functools.singledispatch
+@_not_implemented
+def abs(_):
+    pass
 
 
 @functools.singledispatch
 @_not_implemented
-def abs(_):
+def all(_):
+    pass
+
+
+@functools.singledispatch
+@_not_implemented
+def allclose(_):
+    pass
+
+
+@functools.singledispatch
+@_not_implemented
+def any(_):
     pass
 
 
+arange = np.arange
```

Comment:
How would you do otherwise? The problem is that I want to keep the flexibility of the `np.arange`. If you define it with `functools.singledispatch` then you'll have to provide `start` and `stop`. In practice, often you only use the `stop` argument but sometimes you use both `start` and `stop`. The definition of the API of `np.arange` is quite long in numpy, see https://github.com/numpy/numpy/blob/cdfbdf428d9df9c7119cecae323512a4cd3f57b7/numpy/_core/multiarray.pyi#L712 .

Reply:
And all the functions I directly defined in `core.py` are those that can't be dispatched (like `arange`, `eye`, etc...).